### ISSUE TITLE
Permalinks problem with Page Resources and Ugly URLs 

### ISSUE THREAD

**Issue by mjmurphy** [gohugoio/hugo#4288]
When using the “Page Resources” feature to organise my content there is an issue with uglyURLs.

My content is structured like this:

```
content
    └── posts
        ├── article-title-1.md
        ├── article-title-2.md
        ├── article-title-3
        │   ├── index.md 
        │   └── images
        │       ├── article-image-01.jpg
        │       └── article-image-02.jpg
        └── article-title-4
            ├── index.md
            └── images
                ├── article-image-03.jpg
                └── article-image-04.jpg
```
I have the following in my config.toml
```
uglyURLs = true
canonifyURLs = true
```
When I use the {{ .Permalink }} variable on a single or list page template it does not display the correct full ugly URL.

I would expect the {{ .Permalink }} to output my post urls as:

```
/posts/article-title-1.html
/posts/article-title-2.html
/posts/article-title-3/index.html
/posts/article-title-4/index.html
```
But it’s outputting {{ .Permalink }} as:

```
/posts/article-title-1.html
/posts/article-title-2.html
/posts/article-title-3/
/posts/article-title-4/
```
This means that links to article-title-3 and article-title-4 will not work in my environment.

`Hugo version v0.32.4 windows/amd64`

**Comment by storix** [comment]
Also related to this issue:

The new page resources do not respect the `[permalinks]` setting in `config.toml`. For example, this has no effect:

```
[permalinks]
  posts = ":filename"
```

As for `uglyURLs`, isn't Hugo supposed to output _/posts/article-title-3.html_ instead of _/posts/article-title-3/index.html_? I like the page resources feature, but currently the behaviour is very inconsistent.

**Comment by kaushalmodi** [comment]
@storix I cannot reproduce that (I use page resources but don't set `uglyURLs`).

I have 

```
[Permalinks]
 posts = "/:filename/"
```

And my posts in `content/posts/` show up with intended URL: https://scripter.co/org-contribution-flowchart/

**Comment by storix** [comment]
Sorry to disturb you, but here are exact steps to reproduce the mentioned issue:

- Create a new directory inside content, say music: _content/music/_ 
- Create a new index.md: _content/music/index.md_
- Add this to config.toml 

```
uglyURLs = true
[Permalinks]
  music = "/:filename/"
```
Now Hugo (v0.37 darwin/amd64) will generate _public/music/index.html_, not the expected _public/music.html_. 

Just thought this is the right place to report this issue.

**Comment by kaushalmodi** [comment]
> Create a new index.md: content/music/index.md

You would set permalinks for sections.. here `music` is **not** a section.. it's a *leaf bundle*.. a regular page. See https://gohugo.io/content-management/page-bundles/. Further discussion now definitely needs to happen on Discourse.

**Comment by kf6kjg** [comment]
I'm trying to do something similar - if there's a distinction between my issue and this one, let me know and I'll fork this out into its own issue.

TL;DR: I need to be able to permalink to the name of the branch bundle, not just the name of page bundles.

Our config is similar to the above:
```
[Permalinks]
 products = "/:filename/"
```

Our structure:
```
content/
  products/
    000-000/
      _index.md
      detail1.md
      detail2.md
    111-111/
      _index.md
      detail1.md
  anotherSection/
    ...
```

The goal is to be able to browse to products in the `products` section via a straight URL: e.g. `mysite.com/000-000`

Currently we have Hugo v0.49.2.

I cannot change from `_index` to `index` because my custom theme brings the contents from each detail page and combines them into the list page with the info from the _index. I understand this is a very non-standard setup.

**Comment by speeder** [comment]
Is there are workaround for this?

If not... can someone point me to the correct source file where this is handled? Maybe I can submit a fix (despite never working with go before... but I really, REALLY need this fixed)

**Comment by ulfjack** [comment]
This was originally discussed here:
https://discourse.gohugo.io/t/ugly-urls-with-page-resources/10037/6

Which mentions this workaround:
```
{{ $permalink := cond (strings.HasSuffix .Permalink "html") .Permalink (printf "%sindex.html" .Permalink)
```

This is missing ```}}}``` at the end, but I can confirm that it works.

**Comment by jmooring** [comment]
This was resolved in v0.56.0.

```text
git clone --single-branch -b hugo-github-issue-4288 https://github.com/jmooring/hugo-testing hugo-github-issue-4288
cd hugo-github-issue-4288
hugo server
```
